pages/api/coupon-fraud/claim.js

```diff
@@ -0,0 +1,78 @@
+import { getForbiddenReponse, getOkReponse } from '../../../server/server';
+import { Op } from 'sequelize';
+import { couponEndpoint } from '../../../server/coupon-fraud/coupon-endpoint';
+import { CouponClaim, CouponCode } from '../../../server/coupon-fraud/database';
+
+async function checkVisitorClaimedRecently(visitorId) {
+  const oneHourBefore = new Date();
+  oneHourBefore.setHours(oneHourBefore.getHours() - 1);
+
+  return await CouponClaim.findOne({
+    where: {
+      visitorId,
+      timestamp: {
+        [Op.between]: [oneHourBefore, new Date()],
+      },
+    },
+  });
+}
+
+async function getVisitorClaim(visitorId, couponCode) {
+  return await CouponClaim.findOne({
+    where: { visitorId, couponCode },
+  });
+}
+
+/**
+ * Checks if coupon exists with given coupon code.
+ */
+export async function checkCoupon(code) {
+  return await CouponCode.findOne({
+    where: {
+      code: {
+        [Op.eq]: code,
+      },
+    },
+  });
+}
+
+/**
+ * Claim coupon behalf of visitor
```

Comment:
Please check all the comments you created with some grammar checker, e.g. Grammarly. There are a lot of missing articles and prepositions.

```suggestion
 * Claim coupon on behalf of the visitor.
```